### ISSUE TITLE
The order of convolution and max operation.

### ISSUE THREAD

**Issue by XuyangBai** [WangYueFt/dgcnn#23]
Hi, Thank you for your sharing. I have tried the code of PyTorch version DGCNN and got some questions on your implementation:
1. In `model.py` Why do you first do convolution on x (the return of `get_graph_feature` with dimension `[batch_size, feature_dim * 2, num_points, k]` and then do max operation on the last dimension instead of first calculate the max feature and then do 1d convolution ? Is there any insight on why this order is better ? 
In my understanding, in `get_graph_feature`, we can just return `return torch.max(feature, dim=-1, keepdim=False)[0]` and then in the forward , there is no need to do max pool and we should use `nn.Conv1d` and `nn.BatchNorm1d`
```
        x = get_graph_feature(x, k=self.k)
        x = self.conv1(x)
        x1 = x
```

**Comment by yfyang16** [comment]
@XuyangBai  
I think this kind of idea mainly inherits from pointnet++. The Conv2d can be regarded as a shared MLP to process (feature-x, x). If we directly max(feature-x, x), we may not fully use the correlation of these edges.

**Comment by XuyangBai** [comment]
Hi @yyf710670079  I tried max first and then do the convolution, the final performance is similar, but  I think your explanation is reasonable.

 Thanks a lot.
